CORPSE_deriv.py

```diff
@@ -86,54 +99,89 @@ def CORPSE_deriv(SOM,T,theta,params,claymod=1.0):
-    if isinstance(microbeTurnover,float):
-        microbeTurnover=max(0.0,microbeTurnover)
-    else:
-        microbeTurnover[microbeTurnover<0.0]=0.0
-
-    maintenance_resp=microbeTurnover*(1.0-et)
-    overflow_resp=zeros_like(maintenance_resp)
-
-    deadmic_C_production=microbeTurnover*et   # actual fraction of microbial turnover
-    deadmic_N_production=microbeTurnover*et/params['CN_microbe']
-
-    # C and N available for microbial growth
-    carbon_supply=zeros_like(microbeTurnover)
-    nitrogen_supply=zeros_like(microbeTurnover)
-    for t in chem_types:
-        carbon_supply=carbon_supply+decomp[t+'C']*params['eup'][t]
-        nitrogen_supply=nitrogen_supply+decomp[t+'N']*params['nup'][t]
-
-    IMM_N_max=atleast_1d(params['max_immobilization_rate']*365*SOM['inorganicN']/(SOM['inorganicN']+params['max_immobilization_rate']))
-
-    dmicrobeC=zeros_like(microbeTurnover)
-    dmicrobeN=zeros_like(microbeTurnover)
-    CN_imbalance_term=zeros_like(IMM_N_max)
-
-
-    # Growth is nitrogen limited, with not enough mineral N to support it with max immobilization
-    # loc_Nlim is a vector of True/False that tells the code where this condition applies
-    loc_Nlim=(carbon_supply - maintenance_resp)>((nitrogen_supply+IMM_N_max)*params['CN_microbe'])
-    CN_imbalance_term[loc_Nlim] = -IMM_N_max[loc_Nlim]
-    dmicrobeC[loc_Nlim] =  ((nitrogen_supply[loc_Nlim]+IMM_N_max[loc_Nlim])*params['CN_microbe'] - microbeTurnover[loc_Nlim]*et)
-    dmicrobeN[loc_Nlim] = (nitrogen_supply[loc_Nlim]+IMM_N_max[loc_Nlim] - microbeTurnover[loc_Nlim]*et/params['CN_microbe'])
-    overflow_resp[loc_Nlim]=carbon_supply[loc_Nlim]-maintenance_resp[loc_Nlim] - (nitrogen_supply[loc_Nlim]+IMM_N_max[loc_Nlim])*params['CN_microbe'];
-
-    # Growth must be supported by immobilization of some mineral nitrogen, but is ultimately carbon limited
-    loc_immob=(carbon_supply - maintenance_resp >= nitrogen_supply*params['CN_microbe']) & (carbon_supply - maintenance_resp < (nitrogen_supply+IMM_N_max)*params['CN_microbe'])
-    CN_imbalance_term[loc_immob] = -((carbon_supply[loc_immob]-maintenance_resp[loc_immob])/params['CN_microbe'] - nitrogen_supply[loc_immob])
-    dmicrobeC[loc_immob] = (carbon_supply[loc_immob] - microbeTurnover[loc_immob])
-    dmicrobeN[loc_immob] = ((carbon_supply[loc_immob]-maintenance_resp[loc_immob])/params['CN_microbe'] - microbeTurnover[loc_immob]*et/params['CN_microbe'])
-
-    # Growth is carbon limited and extra N is mineralized
-    loc_Clim=~(loc_Nlim | loc_immob)
-    dmicrobeC[loc_Clim]=(carbon_supply[loc_Clim] - microbeTurnover[loc_Clim]) ;
-    dmicrobeN[loc_Clim]=((carbon_supply[loc_Clim]-maintenance_resp[loc_Clim])/params['CN_microbe'] - microbeTurnover[loc_Clim]*params['et']/params['CN_microbe'])
-    CN_imbalance_term[loc_Clim] = nitrogen_supply[loc_Clim] - (carbon_supply[loc_Clim]-maintenance_resp[loc_Clim])/params['CN_microbe'];
+    microbeTurnover = {'SAP':0.0,'ECM':0.0,'AM':0.0}
```

Comment:
microbeTurnover was previously coded in a way that allows inputs to the model to be an array of multiple values. Setting it as a dictionary of single 0.0 values means you make them one single value and it won't work with multiple values if you want to simulate an array of points simultaneously (using the vector_iterate function, for example).

I'd suggest changing these like this:
```suggestion
    microbeTurnover = {'SAP':zeros_like(SOM['SAPC']),'ECM':zeros_like(SOM['ECMC']),'AM':zeros_like(SOM['AMC'])}
```
Some of these get overwritten below when you set them, but I think it would make a difference for carbon_supply and nitrogen_supply because those get added to the initial value below.

Reply:
Thanks very much for explaining this to me. I see what you mean here and will change them to arrays according to your suggestions here.